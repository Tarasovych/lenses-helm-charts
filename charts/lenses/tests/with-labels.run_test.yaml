suite: enable SAML and check the changes in deployment and secrets
templates:
  - deployment.yaml
  - ingress.yaml
  - service.yaml
tests:
  - it: should work
    set:
      labels:
        foo: bar
      ingress:
        enabled: true
        host: example.com
    asserts:
      # Check the rendered manifest has not changed
      # Review the changes and if they are verified use `helm unittest -u .` to update the manifest
      # Writing tests that cover this chaneg is highly recommended
      - matchSnapshot: {}