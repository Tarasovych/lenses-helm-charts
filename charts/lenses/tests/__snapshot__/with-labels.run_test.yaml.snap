should work:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      annotations: null
      labels:
        app: RELEASE-NAME
        chart: lenses-4.1.4
        component: lenses
        foo: bar
        heritage: Helm
        lenses.io/app: RELEASE-NAME
        lenses.io/app.type: lenses
        release: RELEASE-NAME
      name: RELEASE-NAME
    spec:
      replicas: 1
      selector:
        matchLabels:
          app: RELEASE-NAME
      strategy:
        type: Recreate
      template:
        metadata:
          annotations:
            prometheus.io/path: /metrics
            prometheus.io/port: "9102"
            prometheus.io/scrape: "true"
          labels:
            app: RELEASE-NAME
            component: lenses
            containerPort: "3030"
            foo: bar
            hostPort: "3030"
        spec:
          containers:
          - env:
            - name: LENSES_OPTS
              value: ""
            - name: LICENSE
              valueFrom:
                secretKeyRef:
                  key: license
                  name: RELEASE-NAME
            - name: LENSES_PORT
              value: "3030"
            - name: LENSES_JMX_PORT
              value: "9101"
            - name: LENSES_KAFKA_BROKERS
              value: PLAINTEXT://kafka:9092
            - name: LENSES_SCHEMA_REGISTRY_URLS
              value: ""
            - name: LENSES_SCHEMA_REGISTRY_MODE
              value: ""
            - name: LENSES_KAFKA_CONNECT_CLUSTERS
              valueFrom:
                secretKeyRef:
                  key: connect.clusters
                  name: RELEASE-NAME
            - name: LENSES_ALERT_CONSUMERS_LAG_THRESHOLD
              value: "50000"
            - name: LENSES_ALERT_PLUGINS
              value: ""
            - name: LENSES_TOPICS_AUDITS
              value: _kafka_lenses_audits
            - name: LENSES_TOPICS_METRICS
              value: _kafka_lenses_metrics
            - name: LENSES_TOPICS_PROFILES
              value: _kafka_lenses_profiles
            - name: LENSES_TOPICS_PROCESSORS
              value: _kafka_lenses_processors
            - name: LENSES_TOPICS_CLUSTER
              value: _kafka_lenses_cluster
            - name: LENSES_TOPICS_ALERTS_STORAGE
              value: _kafka_lenses_alerts
            - name: LENSES_TOPICS_ALERTS_SETTINGS
              value: _kafka_lenses_alert_settings
            - name: LENSES_TOPICS_LSQL_STORAGE
              value: _kafka_lenses_lsql_storage
            - name: LENSES_TOPICS_METADATA
              value: _kafka_lenses_topics_metadata
            - name: LENSES_TOPICS_EXTERNAL_TOPOLOGY
              value: __topology
            - name: LENSES_SCHEMA_REGISTRY_TOPICS
              value: null
            - name: LENSES_SQL_EXECUTION_MODE
              value: IN_PROC
            - name: LENSES_SECRET_FILE
              value: /mnt/secrets/security.conf
            - name: LENSES_SCHEMA_REGISTRY_AUTH
              value: ""
            - name: LENSES_SCHEMA_REGISTRY_USERNAME
              value: ""
            - name: LENSES_SCHEMA_REGISTRY_PASSWORD
              valueFrom:
                secretKeyRef:
                  key: schema.registry.password
                  name: RELEASE-NAME
            - name: LENSES_KAFKA_SETTINGS_CLIENT_BASIC_AUTH_CREDENTIALS_SOURCE
              value: ""
            - name: LENSES_KAFKA_SETTINGS_CLIENT_BASIC_AUTH_USER_INFO
              value: ""
            - name: LENSES_KAFKA_SETTINGS_CLIENT_SECURITY_PROTOCOL
              value: PLAINTEXT
            - name: LENSES_KAFKA_SETTINGS_CLIENT_SASL_MECHANISM
              value: GSSAPI
            - name: LENSES_DEPLOYMENTS_EVENTS_BUFFER_SIZE
              value: "10000"
            - name: LENSES_DEPLOYMENTS_ERRORS_BUFFER_SIZE
              value: "1000"
            - name: LENSES_DEPLOYMENT_CONNECT_STATUS_INTERVAL
              value: 30 second
            - name: LENSES_DEPLOYMENT_CONNECT_ACTIONS_BUFFER_SIZE
              value: "1000"
            - name: LT_PACKAGE
              value: helm
            - name: LT_PACKAGE_VERSION
              value: 4.1.4
            image: lensesio/lenses:4.1.3
            imagePullPolicy: IfNotPresent
            livenessProbe:
              httpGet:
                path: /
                port: 3030
              initialDelaySeconds: 300
              periodSeconds: 10
            name: lenses
            ports:
            - containerPort: 3030
            resources:
              limits:
                memory: 5Gi
              requests:
                memory: 4Gi
            volumeMounts:
            - mountPath: /mnt/secrets
              name: secrets
            - mountPath: /etc/keytab
              name: secrets
              subPath: keytab
            - mountPath: /etc/registry.keytab
              name: secrets
              subPath: registry.keytab
            - mountPath: /mnt/settings/security.append.conf
              name: secrets
              subPath: security.append.conf
            - mountPath: /mnt/settings/lenses.append.conf
              name: config
              subPath: lenses.append.conf
          serviceAccountName: default
          volumes:
          - name: secrets
            secret:
              items:
              - key: security.conf
                path: security.conf
              - key: security.append.conf
                path: security.append.conf
              - key: lenses.keytab
                path: lenses.keytab
              - key: keytab
                path: keytab
              - key: client.keystore.jks
                path: client.keystore.jks
              - key: client.truststore.jks
                path: client.truststore.jks
              - key: jaas.conf
                path: jaas.conf
              - key: schema.registry.keystore.jks
                path: schema.registry.keystore.jks
              - key: schema.registry.truststore.jks
                path: schema.registry.truststore.jks
              - key: lenses.opts.keystore.jks
                path: lenses.opts.keystore.jks
              - key: lenses.opts.truststore.jks
                path: lenses.opts.truststore.jks
              secretName: RELEASE-NAME
          - configMap:
              name: RELEASE-NAME
            name: config
  2: |
    apiVersion: extensions/v1beta1
    kind: Ingress
    metadata:
      annotations:
        kubernetes.io/ingress.class: traefik
      labels:
        app: RELEASE-NAME
        chart: lenses-4.1.4
        component: lenses
        foo: bar
        heritage: Helm
        lenses.io/app: RELEASE-NAME
        lenses.io/app.type: lenses-ingress
        release: RELEASE-NAME
      name: RELEASE-NAME
    spec:
      rules:
      - host: example.com
        http:
          paths:
          - backend:
              serviceName: RELEASE-NAME
              servicePort: 80
            path: /
  3: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app: RELEASE-NAME
        chart: lenses-4.1.4
        component: lenses
        foo: bar
        heritage: Helm
        lenses.io/app: RELEASE-NAME
        lenses.io/app.type: lenses-service
        release: RELEASE-NAME
      name: RELEASE-NAME
    spec:
      ports:
      - name: lenses
        port: 80
        targetPort: 3030
      - name: http-metrics
        port: 9102
      selector:
        app: RELEASE-NAME
        component: lenses
      type: ClusterIP
